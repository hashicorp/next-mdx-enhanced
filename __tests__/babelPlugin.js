const path = require('path')
const fs = require('fs')
const { transform } = require('@babel/core')
const plugin = require('../babelPlugin')

test('transforms as intended', () => {
  const mockNextOptions = { dir: __dirname }
  const { content, filename } = loadFixture('basic')
  const { code } = transform(content, {
    plugins: [plugin(mockNextOptions)],
    filename
  })
  expect(code).toMatchSnapshot()
})

function loadFixture(name) {
  const filename = path.join(__dirname, `fixtures/babel/${name}.js`)
  return { content: fs.readFileSync(filename, 'utf8'), filename }
}
