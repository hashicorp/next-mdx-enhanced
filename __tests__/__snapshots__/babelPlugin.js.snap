// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms as intended 1`] = `
"import something from \\"PATH/TO/NEXT/PROJECT/.next/cache/mdx-data/9edc110b0a5f0cf4742677e8539f30c9.json\\";
import a, { c as d } from './foo.mdx';
import b from \\"PATH/TO/NEXT/PROJECT/.next/cache/mdx-data/9edc110b0a5f0cf4742677e8539f30c9.json\\";"
`;
